lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -71,6 +71,7 @@ def serializable_hash_for_single_resource(options)
         end
 
         def resource_identifier_type_for(serializer)
+          return serializer._type if serializer._type.present?
```

Comment:
I'm not crazy about `present?` at all and totally ok with having just _type. I just find implicit type-conversion confusing and try to avoid that when possible.